client/miner.c

```diff
@@ -197,6 +300,22 @@ void *miner_net_thread(void *arg)
 		pthread_mutex_unlock(&g_miner_mutex);
 		goto err;
 	}
+
+	if(g_multi_address){
```

Comment:
I think it is not a good idea to start the thread here. I suggest you to do it in xdag_initialize_miner function

Reply:
Good suggestion, I will move to xdag_initialize_miner